movement/kinematics.py

```diff
@@ -324,6 +326,330 @@ def compute_head_direction_vector(
+
+        - If a dictionary is provided,
+          the key is the keypoint or individual and
+          the value can be a list of keypoints or individuals,
+          or a string representing a single keypoint or individual
+          to compute the distance with.
+        - If the special keyword ``'all'`` is provided,
+          all possible combinations of pairs are computed.
+    metric : str, optional
+        The distance metric to use. Must be one of the options supported
+        by :func:`scipy.spatial.distance.cdist`, e.g. ``'cityblock'``,
+        ``'euclidean'``, etc.
+        Defaults to ``'euclidean'``.
+    **kwargs : dict
+        Additional keyword arguments to pass to
+        :func:`scipy.spatial.distance.cdist`.
+
+    Returns
+    -------
+    xarray.DataArray | dict[str, xarray.DataArray]
+        An :class:`xarray.DataArray` containing the computed distances between
+        a single pair of ``individuals`` or ``keypoints``. If multiple
+        ``pairs`` are provided, this will be a dictionary containing the
+        computed distances for each pair, with the key being the pair of
+        keypoints or individuals and the value being the
+        :class:`xarray.DataArray` containing the computed distances.
+
+    Raises
+    ------
+    ValueError
+        If ``dim`` is not one of ``'individuals'`` or ``'keypoints'``;
+        if ``pairs`` is not a dictionary or ``'all'``; or
+        if there are no pairs in ``data`` to compute distances for.
+
+    Examples
+    --------
+    Compute the Euclidean distance (default) for all keypoints
+    between ``ind1`` and ``ind2`` (i.e. interindividual distance):
+
+    >>> position = xr.DataArray(
+    ...     np.arange(36).reshape(2, 3, 3, 2),
+    ...     coords={
+    ...         "time": np.arange(2),
+    ...         "individuals": ["ind1", "ind2", "ind3"],
+    ...         "keypoints": ["key1", "key2", "key3"],
+    ...         "space": ["x", "y"],
+    ...     },
+    ...     dims=["time", "individuals", "keypoints", "space"],
+    ... )
+    >>> dist_ind1_ind2 = compute_pairwise_distances(
+    ...     position, "individuals", {"ind1": "ind2"}
+    ... )
+    >>> dist_ind1_ind2
+    <xarray.DataArray (time: 2, ind1: 3, ind2: 3)> Size: 144B
+    8.485 11.31 14.14 5.657 8.485 11.31 ... 5.657 8.485 11.31 2.828 5.657 8.485
+    Coordinates:
+    * time     (time) int64 16B 0 1
+    * ind1     (ind1) <U4 48B 'key1' 'key2' 'key3'
+    * ind2     (ind2) <U4 48B 'key1' 'key2' 'key3'
+
+    The resulting ``dist_ind1_ind2`` is a DataArray containing the computed
+    distances between ``ind1`` and ``ind2`` for all keypoints
+    at each time point.
+
+    To obtain the distances between ``key1`` of ``ind1`` and
+    ``key2`` of ``ind2``:
+
+    >>> dist_ind1_ind2.sel(ind1="key1", ind2="key2")
+
+    Compute the Euclidean distance (default) between ``key1`` and ``key2``
+    for all pairs of individuals and within each individual
+    (i.e. interkeypoint distance):
```

Comment:
If you adopt the previous suggestion, then accept this one two (to exactly match phrasing between the two examples)

```suggestion
    Compute the Euclidean distance (default) between ``key1`` and ``key2``
    (i.e. interkeypoint distance), for all possible pairs of individuals.
```